libscript/src/script-instance.cpp

```diff
@@ -1369,6 +1161,316 @@ __MCScriptResolveForeignFunctionBindingForJava(MCScriptInstanceRef p_instance,
     return true;
 }
 
+bool MCScriptForeignHandlerInfoParse(MCStringRef p_binding, MCScriptForeignHandlerInfo*& r_info)
+{
+    MCScriptForeignHandlerInfo* t_info = new MCScriptForeignHandlerInfo;
```

Comment:
I think there are some leaks here... The t_info instance will get leaked if there is an early exit due to an error - similarly t_rest will also leak a reference in that case.

Reply:
Woah... sorry leaking like a sieve sorry... even on success it's leaking the returned struct pointer because `MCScriptForeignHandlerInfoRelease` doesn't delete it.